### PR TITLE
Lots of TPL changes

### PR DESCRIPTION
**Pull request by white238**
This was beginning to be scattered across 3 branches.  Pulling them all into one.

### CODE REVIEW

File: cmake/thirdparty/SetupSeracThirdParty.cmake [hunk 1]
```diff
@@ -66,6 +66,8 @@ if (NOT SERAC_THIRD_PARTY_LIBRARIES_FOUND)
     #------------------------------------------------------------------------------
     # Caliper
     #------------------------------------------------------------------------------
+    # TODO: temporarily turned off until PR # 626
+    set(CALIPER_DIR "" CACHE STRING "" FORCE)
```
Comment (white238) [inline cmake/thirdparty/SetupSeracThirdParty.cmake:70]: @adayton1 if this gets in before your branch. Please undo this line.

File: src/serac/infrastructure/accelerator.hpp [hunk 2]
```diff
@@ -145,6 +147,19 @@ using GPUArray = ExecArray<T, dim, ExecutionSpace::GPU>;
 template <typename T, int dim = 1>
 using UnifiedArray = ExecArray<T, dim, ExecutionSpace::Dynamic>;
 
+#else
+// If not a CUDA build then force all arrays to be CPU
```
Comment (white238) [inline src/serac/infrastructure/accelerator.hpp:151]: There may be a better way of handling this but it is late and I have dreams of disney.

**Comment by samuelpmishLLNL** [comment]
I'm still unable to build serac with CUDA support when using this branch. When I invoke `uberenv`,
```
python3 scripts/uberenv/uberenv.py --prefix=../serac_libs_cuda --spack-config-dir=scripts/spack/configs/linux_ubuntu_20/ --spec "+cuda cuda_arch=61"
```
the TPLs finish installing, but when configuring cmake, I get the following error message:
```
[proc] Executing command: /usr/bin/cmake --no-warn-unused-cli -DSERAC_ENABLE_CODEVELOP=ON -C /home/sam/code/serac/.vscode/serac_libs_cuda.cmake -DENABLE_DOCS=ON -DENABLE_CLANGFORMAT=ON -DCLANGFORMAT_EXECUTABLE=/usr/bin/clang-format -DSERAC_ENABLE_CODE_CHECKS=ON -DSPHINX_EXECUTABLE=/usr/bin/sphinx-build -DDOXYGEN_EXECUTABLE=/usr/bin/doxygen -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/bin/x86_64-linux-gnu-gcc-9 -DCMAKE_CXX_COMPILER:FILEPATH=/bin/x86_64-linux-gnu-g++-9 -H/home/sam/code/serac -B/home/sam/code/serac/build -G Ninja
[cmake] Not searching for unused variables given on the command line.
[cmake] loading initial cache file /home/sam/code/serac/.vscode/serac_libs_cuda.cmake
[cmake] -- The CXX compiler identification is GNU 9.3.0
[cmake] -- The C compiler identification is GNU 9.3.0
[cmake] -- Detecting CXX compiler ABI info
[cmake] -- Detecting CXX compiler ABI info - done
[cmake] -- Check for working CXX compiler: /bin/x86_64-linux-gnu-g++-9 - skipped
[cmake] -- Detecting CXX compile features
....
[cmake] -- Removing non-existant include directories from target[conduit_relay_mpi]
[cmake] -- Removing non-existant include directories from target[conduit_relay_mpi_io]
[cmake] -- Removing non-existant include directories from target[conduit_blueprint]
[cmake] -- Removing non-existant include directories from target[conduit_blueprint_mpi]
[cmake] -- Configuring Serac version v0.0.1
[cmake] -- Serac Code Checks: ON
[cmake] -- Configuring done
[cmake] CMake Error: install(EXPORT "serac-targets" ...) includes target "axom" which requires target "openmp" that is not in this export set, but in multiple other export sets: lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake.
[cmake] An exported target cannot depend upon another target which is exported multiple times. Consider consolidating the exports of the "openmp" target to a single export.
[cmake] CMake Error: install(EXPORT "serac-targets" ...) includes target "axom" which requires target "openmp" that is not in this export set, but in multiple other export sets: lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake.
[cmake] An exported target cannot depend upon another target which is exported multiple times. Consider consolidating the exports of the "openmp" target to a single export.
[cmake] CMake Error: install(EXPORT "serac-targets" ...) includes target "axom" which requires target "openmp" that is not in this export set, but in multiple other export sets: lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake.
[cmake] An exported target cannot depend upon another target which is exported multiple times. Consider consolidating the exports of the "openmp" target to a single export.
[cmake] CMake Error: install(EXPORT "serac-targets" ...) includes target "axom" which requires target "openmp" that is not in this export set, but in multiple other export sets: lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake, lib/cmake/axom-targets.cmake.
[cmake] An exported target cannot depend upon another target which is exported multiple times. Consider consolidating the exports of the "openmp" target to a single export.
[cmake] -- Generating done
[cmake] CMake Generate step failed.  Build files cannot be regenerated correctly.
```
this is with my native ubuntu 20.04 machine

**Review by jamiebramwell — APPROVED** [review]

**Review by samuelpmishLLNL — APPROVED** [review]

**Comment by jamiebramwell** [comment]
@adayton1 @samuelpmishLLNL could you take a look at this? I disabled the sundials test while we triage the intermittent error so we can merge this and https://github.com/LLNL/serac/pull/599 .

**Comment by jamiebramwell** [comment]
Ha! Nevermind! You approved it @samuelpmishLLNL ! 😆

**Pull request merged**